src/engines/kotorbase/module.cpp

```diff
@@ -179,6 +180,26 @@ Creature *Module::getPC() {
 	return _pc;
 }
 
+void Module::switchPC(unsigned int index) {
+	if (!_partyController.isAvailableCreature(index))
+		throw Common::Exception("%u is not an available npc id", index);
+
+	Creature *newPC = createCreature(_partyController.getAvailableNPCTemplate(index));
+	ObjectMan.registerObject(newPC);
+	_partyController.addPartyMember(index, newPC);
+	int oldLeaderID = _partyController.getPartyLeaderID();
+
+	Creature *oldPC = _partyController.getPartyLeader();
+	_partyController.setPartyLeader(index);
+	_partyController.removePartyMember(oldLeaderID);
+	_pc = newPC;
+	notifyPartyLeaderChanged();
+
+	_area->removeObject(oldPC);
+	removeObject(*oldPC);
```

Comment:
`Module::removeObject` is called implicitly in `Area::removeObject`. See https://github.com/xoreos/xoreos/blob/90003390a2c8b908335e90517e6580142e3b4a49/src/engines/kotorbase/area.cpp#L872